hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/block/SCMBlockDeletingService.java

```diff
@@ -143,10 +147,14 @@ public EmptyTaskResult call() throws Exception {
             return EmptyTaskResult.newResult();
           }
 
+          Set<Long> processedTxIDs = new HashSet<>();
           for (Map.Entry<UUID, List<DeletedBlocksTransaction>> entry :
               transactions.getDatanodeTransactionMap().entrySet()) {
             UUID dnId = entry.getKey();
             List<DeletedBlocksTransaction> dnTXs = entry.getValue();
+            processedTxIDs.addAll(dnTXs.stream()
```

Comment:
Wouldn't it be better to add this inside `if (!dnTXs.isEmpty())`, So that we don't have this on empty lists?